homeassistant/components/hunterdouglas_powerview/scene.py

```diff
@@ -2,86 +2,73 @@
 import logging
 from typing import Any
 
-from aiopvapi.helpers.aiorequest import AioRequest
 from aiopvapi.resources.scene import Scene as PvScene
-from aiopvapi.rooms import Rooms
-from aiopvapi.scenes import Scenes
 import voluptuous as vol
 
-from homeassistant.components.scene import DOMAIN, Scene
-from homeassistant.const import CONF_PLATFORM
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.components.scene import Scene
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_HOST, CONF_PLATFORM
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.entity import async_generate_entity_id
+
+from .const import (
+    COORDINATOR,
+    DEVICE_INFO,
+    DOMAIN,
+    HUB_ADDRESS,
+    PV_API,
+    PV_ROOM_DATA,
+    PV_SCENE_DATA,
+    ROOM_NAME_UNICODE,
+    STATE_ATTRIBUTE_ROOM_NAME,
+)
+from .entity import HDEntity
 
 _LOGGER = logging.getLogger(__name__)
-ENTITY_ID_FORMAT = DOMAIN + ".{}"
-HUB_ADDRESS = "address"
 
 PLATFORM_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_PLATFORM): "hunterdouglas_powerview",
-        vol.Required(HUB_ADDRESS): cv.string,
-    }
+    {vol.Required(CONF_PLATFORM): DOMAIN, vol.Required(HUB_ADDRESS): cv.string}
 )
 
 
-SCENE_DATA = "sceneData"
-ROOM_DATA = "roomData"
-SCENE_NAME = "name"
-ROOM_NAME = "name"
-SCENE_ID = "id"
-ROOM_ID = "id"
-ROOM_ID_IN_SCENE = "roomId"
-STATE_ATTRIBUTE_ROOM_NAME = "roomName"
-
+def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
We need to use the async api here since we want to call the async api inside.

Reply:
I missed it again. 🤦

Reply:
https://github.com/home-assistant/core/pull/34899